### PR TITLE
Program

### PR DESCRIPTION
**Pull request by PoojaLad55**
Migrated programs page

### CODE REVIEW

File: src/app/programs/page.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    description: string;
+  }
+
+  const programRequest = await fetch("http://localhost:3000/api/programs");
+  const programs: Program[] = await programRequest.json();
+
+  return (
+    <div className="bg-zinc-900 min-h-screen px-16 py-12 bg-opacity-95">
+      <h1 className="text-4xl font-bold mb-3 text-white">Programs</h1>
+      <div className="grid grid-cols-3 gap-6 lg:grid-cols-4">
```
Comment (02danieljonas) [inline src/app/programs/page.tsx:17]: On smaller displays, the grid should only have one column: `sm:grid-cols-1 md:grid-cols-3`

File: src/app/programs/programCard.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+              xmlns="http://www.w3.org/2000/svg"
+              viewBox="0 0 1200 480.12"
+            >
+              <path d="m499.17,75.78c-79.68,0-131.34,62.17-153.23,108.58-3.5,7.01-7.88,17.51-18.39,13.13-11.38-4.38-2.63-21.89,4.38-35.9,7-14.01,11.38-28.9,11.38-49.91v-21.89h0c-77.37,0-140.1,62.73-140.1,140.1v250.23h140.1l1.02-163.32c0-87.13,37.48-123.43,94.39-123.43,49.91,0,86.72,30.4,86.72,134.6v152.16h140.1v-208.2c0-131.34-66.55-196.14-166.37-196.14Z" />
+              <path d="m1140.2,201.87c33.02,0,59.8-26.77,59.8-59.8v-52.28h-126.09V0h-140.1v338.26c0,111.2,43.78,141.85,136.6,141.85h129.59v-112.08h-84.06c-40.28,0-42.03-1.75-42.03-45.53v-120.64h66.29Z" />
+              <path d="m800.09,142.42c-27,0-51.55-12.01-69.82-31.6v369.3h138.95V111.55c-18.19,19.17-42.47,30.87-69.13,30.87Z" />
+              <path d="m799.86,116.1c31.38,0,56.82-25.49,56.82-56.93s-25.44-56.93-56.82-56.93-56.82,25.49-56.82,56.93,25.44,56.93,56.82,56.93Z" />
+              <path d="m69.82,142.42c-27,0-51.55-12.01-69.82-31.6v369.3h138.95V111.55c-18.19,19.17-42.47,30.87-69.13,30.87Z" />
+              <ellipse cx="69.59" cy="59.18" rx="56.82" ry="56.93" />
+            </svg>
```
Comment (02danieljonas) [inline src/app/programs/programCard.tsx:41]: You can add the SVG to a separate file for a cleaner finish: 
https://github.com/INIT-Build-Web-Dev-3-Spring-2024/init-website/blob/b712030bc40b4e0259ac07227f0f579b65846980/src/components/Navbar.tsx#L9
https://github.com/INIT-Build-Web-Dev-3-Spring-2024/init-website/blob/b712030bc40b4e0259ac07227f0f579b65846980/src/components/Navbar.tsx#L63
[InitLogo](https://github.com/INIT-Build-Web-Dev-3-Spring-2024/init-website/blob/b712030bc40b4e0259ac07227f0f579b65846980/public/assets/images/logo.svg)

File: src/app/programs/page.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+export default async function ProgramsPage() {
+  interface Program {
+    name: string;
+    color: string;
+    image: string;
+    description: string;
+  }
+
+  const programRequest = await fetch("http://localhost:3000/api/programs");
```
Comment (styled-man) [inline src/app/programs/page.tsx:11]: This being hardcoded will give an error when someone is not running the app on localhost:3000.

Use this solution from the job branch that uses a [middleware.ts](https://github.com/INIT-Build-Web-Dev-3-Spring-2024/init-website/blob/job/src/middleware.ts) file and a [dynamic fetch url](https://github.com/INIT-Build-Web-Dev-3-Spring-2024/init-website/blob/c60f88c375cfbef65849c73900df69094882fb3d/src/app/jobs/%40companies/page.tsx#L12)

File: src/app/api/programs/route.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+      description:
+        "the hardware development side of technology. Using all sorts of machinery, large or small, this program aims inspire members to create their own devices with the technology available today.",
+    },
+    {
+      name: "Build",
+      color: "#FC6C3F",
+      image: "/images/programs/initbuild.webp",
+      description:
+        "project development. Teams have nine weeks to complete a project corresponding to a topic in technology that they may be interested in. From mobile applications to games to even robots.",
+    },
```
Comment (02danieljonas) [inline src/app/api/programs/route.ts:23]: You modified these description to work best within your code.

File: src/app/api/programs/route.ts [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+      description:
+        "INIT Uplift is a program focused on mentorship. Many incoming freshmen are overwhelmed by the amount of information available to them from the start. Our program aims to guide these new students and help them kickstart their tech careers. Those with experience in the field or program help to navigate freshmen during these rough times and allow them to establish a network or two. Mentor with us or get a mentor today",
+    },
+    {
+      name: "Launch",
+      color: "#11CEBB",
+      image: "/images/programs/initlaunch.webp",
+      description:
+        "INIT Launch is a program focused on project development. Teams have nine weeks to complete a project corresponding to a topic in technology that they may be interested in. From mobile applications to games to even robots, this program aims teach you the fundamentals of project development in the real world. No experience is required to spark up your portfolio with us.",
+    },
```
Comment (02danieljonas) [inline src/app/api/programs/route.ts:58]: You did not modify these descriptions and the output looks like: `INIT Discover is a program focused on INIT Discover is a program focused...`

File: src/app/api/programs/route.ts
Comment (02danieljonas) [inline src/app/api/programs/route.ts:1]: If you have to modify the description, you can remove the `INIT Explore` but keep the `is a program focused on`, so it is not static within your code: 
https://github.com/INIT-Build-Web-Dev-3-Spring-2024/init-website/blob/11a055b88de5e44da568d3c42fc7ccfbe37bb854/src/app/programs/programCard.tsx#L47-L48

**Review by 02danieljonas — CHANGES_REQUESTED** [review]

**Review by styled-man — CHANGES_REQUESTED** [review]

**Review by styled-man — APPROVED** [review]

**Review by styled-man — APPROVED** [review]

**Review by KGCarol — APPROVED** [review]

**Review by 02danieljonas — APPROVED** [review]
LGTM!

**Pull request merged**